apps/dashboard/features/token-distribution/components/TokenDistributionChart.tsx

```diff
@@ -0,0 +1,147 @@
+
+  const datasets = appliedMetrics.reduce(
+    (acc, key) => {
+      acc[key] = timeSeriesData[key];
+      return acc;
+    },
+    {} as Record<MetricTypesEnum, DaoMetricsDayBucket[]>,
+  );
+
+  const allDates = new Set(
+    Object.values(datasets).flatMap((dataset) =>
+      dataset?.map((item) => item.date),
+    ),
+  );
+
+  const chartData = Array.from(allDates)
+    .sort((a, b) => Number(a) - Number(b))
+    .map((date) => {
+      const dataPoint: Record<string, number | null> = {
+        date: Number(date),
+      };
+
+      Object.keys(datasets).forEach((key) => {
+        const entry = datasets[key as keyof typeof datasets]?.find(
+          (item) => item.date === date,
+        );
+        dataPoint[key] = entry ? Number(entry.high) / 1e18 : null;
+      });
+
+      return dataPoint;
+    })
+    .filter(
+      (dataPoint) => !Object.values(dataPoint).some((value) => value == null),
+    );
+
+  const isMocked = Object.values(datasets).every(
+    (value) => value!.length === 0,
+  );
+
+  return (
+    <div className="border-light-dark bg-surface-default text-primary relative flex h-[300px] w-full flex-col items-center justify-center rounded-lg">
+      {isMocked && <ResearchPendingChartBlur />}
+      <ChartContainer
+        className="h-full w-full justify-start"
+        config={chartConfig}
+      >
+        <ComposedChart data={chartData}>
+          <CartesianGrid vertical={false} stroke="#27272a" />
+          <XAxis
+            dataKey="date"
+            scale="time"
+            type="number"
+            domain={["auto", "auto"]}
+            tickMargin={8}
+            tickFormatter={(date) => timestampToReadableDate(date)}
+          />
+          <YAxis
+            domain={["auto", "auto"]}
+            tickFormatter={(value) => formatNumberUserReadable(Number(value))}
+          />
+          <Tooltip
+            content={
+              <TokenDistributionCustomTooltip chartConfig={chartConfig} />
+            }
+          />
+          {/* <Line dataKey="VERTICAL_LINE_EXAMPLE" stroke="#8884d8" /> */}
+          {/* <Bar dataKey="BAR_EXAMPLE" stroke="#8884d8" /> */}
+          {Object.keys(chartConfig).map((key) => {
+            const isOpaque = hoveredMetricKey && !(key === hoveredMetricKey);
+            return (
+              <Line
+                key={key}
+                dataKey={key}
+                stroke={chartConfig[key as keyof typeof chartConfig].color}
+                strokeWidth={2}
+                strokeOpacity={isOpaque ? 0.3 : 1}
+                dot={false}
+              />
+            );
+          })}
+          <Brush
+            dataKey="date"
+            height={32}
+            stroke="#333"
+            fill="#1f1f1f"
+            tickFormatter={(timestamp) => timestampToReadableDate(timestamp)}
+          >
+            <AreaChart height={32} width={1128} data={chartData}>
+              <XAxis dataKey="date" hide />
+              <YAxis hide />
+              <Area
```

Comment:
The TokenDistributionChart component accesses appliedMetrics[0] without checking if the array is empty. This will cause a runtime error when appliedMetrics is an empty array, breaking the Brush component's Area dataKey.